cpp/ql/test/query-tests/Security/CWE/CWE-676/semmle/PotentiallyDangerousFunction/test.c

```diff
@@ -36,7 +36,7 @@ char *gets(char *s);
 
 void testGets() {
 	char buf1[1024];
-	char buf2 = malloc(1024);
+	char *buf2 = malloc(1024);
```

Comment:
Wait... How did this ever compile? a "void pointer to char" conversion doesn't sound like something that C would do implicitly?

Reply:
Probably because it will implicitly define a function `gets(char)`.